portal/src/Widget.tsx

```diff
@@ -1,23 +1,80 @@
-import React from "react";
+import React, {
+  useState,
+  useCallback,
+  useLayoutEffect,
+  useRef,
+  useMemo,
+} from "react";
+import { IconButton, DefaultEffects } from "@fluentui/react";
 import cn from "classnames";
-import { DefaultEffects } from "@fluentui/react";
+
 import styles from "./Widget.module.css";
 
-export interface WidgetProps {
+interface WidgetProps {
   className?: string;
+  extendable?: boolean;
   children?: React.ReactNode;
 }
 
+const ICON_PROPS = {
+  iconName: "ChevronDown",
+};
+
+// 16px top padding + 32px icon button height + 16px bottom padding
+const COLLAPSED_HEIGHT = 64;
+
 const Widget: React.VFC<WidgetProps> = function Widget(props: WidgetProps) {
-  const { className, children } = props;
+  const { className, extendable, children } = props;
+  const [extended, setExtended] = useState(extendable ?? true);
```

Comment:
Currently, when `extendable` is empty, it implies the widget is non-extendable. And the `true` or `false` value of `extendable` is the initial value of `extended`? I think it is a bit confusing.